src/main/java/com/amazonaws/services/kinesis/clientlibrary/lib/worker/KinesisDataFetcher.java

```diff
@@ -130,19 +148,25 @@ public boolean isShardEnd() {
         }
     }
 
+    private boolean isValidResult(GetRecordsResult getRecordsResult) {
+        return getRecordsResult.getNextShardIterator() == null ? !CollectionUtils.isNullOrEmpty(getRecordsResult.getChildShards())
+                                                    : getRecordsResult.getNextShardIterator() != null && getRecordsResult.getChildShards().isEmpty();
```

Comment:
The `getRecordsResult.getNextShardIterator() != null` is redundant, but we can keep it if you think it improves code clarity

Reply:
Ah, yeah, it's redundant